src/compile/scale/domain.ts

```diff
@@ -79,10 +79,15 @@ export default function domain(scale: Scale, model: Model, channel:Channel): any
       if (channel === 'x' || channel === 'y') {
         // X/Y position have to include start and end for non-ordinal scale
         return {
-          data: model.dataTable(),
+          // FIXME: use unioned domain where the data is defined once and fields is only an array of strings
```

Comment:
I assume this breaks when the original `{data: ..., fields: ...}` format need to get merge with others?  

If so I wonder if we should modify the merge logic to be responsible for converting `{data: D, fields: [A,B]}` into `{fields: [{data: D, field: A},{data: D, field: B}]}` instead of modifying it here (and thus you can remove FIXME).  

(This should be relatively simple, so I guess we don't have to punt on this.)